framework/src/Volo.Abp.Caching/Volo/Abp/Caching/DistributedCache.cs

```diff
@@ -172,33 +193,57 @@ protected virtual void SetDefaultOptions()
             {
                 return GetManyFallback(
                     keyArray,
+                    considerUow,
                     hideErrors
                 );
             }
 
+            var cachedValues = new List<KeyValuePair<TCacheKey, TCacheItem>>();
+            var notCachedKeys = new List<TCacheKey>();
+            if (ShouldConsiderUow(considerUow))
+            {
+                var cache = GetUnitOfWorkCache();
+                foreach (var key in keyArray)
+                {
+                    var value = cache.GetOrDefault(key)?.GetUnRemovedValueOrNull();
+                    if (value != null)
+                    {
+                        cachedValues.Add(new KeyValuePair<TCacheKey, TCacheItem>(key, value));
+                    }
+                }
+
+                notCachedKeys = keyArray.Except(cachedValues.Select(x => x.Key)).ToList();
+                if (!notCachedKeys.Any())
+                {
+                    return cachedValues.ToArray();
+                }
+            }
+
             hideErrors = hideErrors ?? _distributedCacheOption.HideErrors;
             byte[][] cachedBytes;
 
+            var readKeys = notCachedKeys.Any() ? notCachedKeys.ToArray() : keyArray;
             try
             {
-                cachedBytes = cacheSupportsMultipleItems.GetMany(keyArray.Select(NormalizeKey));
+                cachedBytes = cacheSupportsMultipleItems.GetMany(readKeys.Select(NormalizeKey));
```

Comment:
I didn't understand that. It seems the method is returning only the notCachedKeys instead of all keys in this case. Shouldn't it return the values of all the keys?

Reply:
Yes, I revert the changes, it now returns `keyArray`

Reply:
The logic here is:
First read the keys in the uow cache, of course, some keys may not be in the uow cache, then read the remaining keys from the cache, and finally merge the results.

```cs
return cachedValues.Concat(ToCacheItems(cachedBytes, readKeys)).ToArray();
```